src/net_processing.cpp

```diff
@@ -2136,7 +2136,7 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
             BOOST_FOREACH(uint256 hash, vEraseQueue)
                 EraseOrphanTx(hash);
         }
-        else if (!AlreadyHave(inv) && tx.IsZerocoinSpend() && !tx.IsSigmaSpend() && !tx.IsLelantusJoinSplit() && AcceptToMemoryPool(mempool, state, ptx, true, &fMissingInputsSigma, nullptr, false, 0, true)) {
+        else if (!AlreadyHave(inv) && !tx.IsSigmaSpend() && !tx.IsLelantusJoinSplit() && AcceptToMemoryPool(mempool, state, ptx, true, &fMissingInputsSigma, nullptr, false, 0, true)) {
             // Changes to mempool should also be made to Dandelion stempool
```

Comment:
I believe in this scope we are working with newly broadcasted transactions, as we will not have such, better to remove it, 
but anyway in in terms of just conditions almost nothing is changed, if tx is ZerocoinSpend() all other conditions are true, it will call AcceptToMemoryPool and go inside if.